bpf-sys/src/headers.rs

```diff
@@ -61,8 +61,16 @@ pub fn running_kernel_version() -> Option<String> {
     })
 }
 
-pub fn build_kernel_version() -> Result<KernelVersion, Box<dyn Error>> {
-    let KernelHeaders { source: _, build } = kernel_headers_path()?;
+pub fn build_kernel_version(source_dir: Option<&Path>) -> Result<KernelVersion, Box<dyn Error>> {
```

Comment:
I think it would be better if this version was called `build_kernel_version_from`, and the original `build_kernel_version` function provided a default wrapper. This means we can avoid breaking API, and the resulting code will also be cleaner.